tests/mapping.test.ts

```diff
@@ -0,0 +1,274 @@
+import {
+  clearStore,
+  test,
+  assert,
+  logStore,
+} from "matchstick-as/assembly/index";
+import { Address, BigInt } from "@graphprotocol/graph-ts";
+import { handleTransfer, NON_CIRCULATING, COW_TOKEN } from "../src/mapping";
+import { mockTransferEvent, handleMultipleTransfers } from "./utils";
+import { Holder, Supply } from "../generated/schema";
+
+const REGULAR_ACCOUNT = [
+  Address.fromString("0x0000000000000000000000000000000000000001"),
+  Address.fromString("0x0000000000000000000000000000000000000002"),
+  Address.fromString("0x0000000000000000000000000000000000000003"),
+];
+
+const TWO = BigInt.fromI32(2);
+
+class TestStore {
+  holders: Holder[];
+  supply: Supply;
+}
+
+function setupStore(): TestStore {
+  /// Two Mint 2000 tokens with half going to a regular account
+  //  and the other half in a non-circulating account: Store Results in
+  // {
+  //   "Holder": {
+  //     "0x0000000000000000000000000000000000000001": {
+  //       "balance": {
+  //         "type": "BigInt",
+  //         "data": "1000"
+  //       },
+  //       "id": {
+  //         "type": "String",
+  //         "data": "0x0000000000000000000000000000000000000001"
+  //       }
+  //     },
+  //     "0xd057b63f5e69cf1b929b356b579cba08d7688048": {
+  //       "id": {
+  //         "type": "String",
+  //         "data": "0xd057b63f5e69cf1b929b356b579cba08d7688048"
+  //       },
+  //       "balance": {
+  //         "type": "BigInt",
+  //         "data": "1000"
+  //       }
+  //     }
+  //   },
+  //   "Supply": {
+  //     "0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB": {
+  //       "circulating": {
+  //         "type": "BigInt",
+  //         "data": "1000"
+  //       },
+  //       "id": {
+  //         "type": "String",
+  //         "data": "0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB"
+  //       },
+  //       "total": {
+  //         "type": "BigInt",
+  //         "data": "2000"
+  //       }
+  //     }
+  //   }
+  // }
+  const amount = BigInt.fromI32(1000);
+  let transferEvents = [
+    mockTransferEvent(Address.zero(), REGULAR_ACCOUNT[0], amount, "0x"),
+    mockTransferEvent(Address.zero(), NON_CIRCULATING[0], amount, "0x"),
+  ];
+  handleMultipleTransfers(transferEvents);
+
+  const holder1 = new Holder(REGULAR_ACCOUNT[0].toHex());
+  holder1.balance = amount;
+  const holder2 = new Holder(NON_CIRCULATING[0].toHex());
+  holder2.balance = amount;
+
+  const supply = new Supply(COW_TOKEN);
+  supply.circulating = amount;
+  supply.total = amount.times(TWO);
+
+  return {
+    holders: [holder1, holder2],
+    supply,
+  };
```

Comment:
I find this pattern really strange: first generating a state and then duplicating the state in variables + testing that the duplication was correct.

Why can't we just generate a state and then read that state?

Reply:
I just didn't want to duplicate the setup over and over again in the tests (i.e removing code redundancy). It doesn't have to be this way, but the setup is well tested so it seems like a good approach to me.

Reply:
We can't read the state because the only way to access the fields with this framework is via strings. It would not be very nice at all. What I am doing here is the most natural way of having what we need.